### PR TITLE
Adding Course Enrolments api endpoint

### PR DESCRIPTION
**Pull request by madnihamza1841**
**Description:** 
An endpoint for course_enrollments has been added using an existing view from figures. 
URL of endpoint: `/api/edly_panel/v1/courses/course_enrollment/`

**JIRA:** 
https://edlyio.atlassian.net/browse/EDLY-4356

**Testing instructions:**

1. Open page A
2. Do thing B
3. Expect C to happen
4. If D happened instead - check failed.

**Merge checklist:**

- [ ] All reviewers approved
- <del>[ ] CI build is green
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are (reasonably) squashed
- [ ] Translations are updated

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

**Note:** Please add screenshots for any viewable change.

### CODE REVIEW

File: lms/urls.py [hunk 1]
```diff
@@ -172,6 +172,8 @@
     url(r'^api/experiments/', include(('experiments.urls', 'lms.djangoapps.experiments'), namespace='api_experiments')),
     url(r'^api/discounts/', include(('openedx.features.discounts.urls', 'openedx.features.discounts'),
                                     namespace='api_discounts')),
+
+    url(r'^api/v1/courses/', include('openedx.features.edly.api.urls'))
```
Comment (taimoor-ahmed-1) [inline lms/urls.py:176]: add comma to reduce git difference.

**Review by burhandodhy — COMMENTED** [review]
@madnihamza1841 you PR is not fulfilling these requirements 

API endpoint should be accessible for EDLY_API_USERS_GROUP
Add unit tests

cc: @taimoor-ahmed-1

File: lms/urls.py (hunk 1 above)
Comment (madnihamza1841) [reply lms/urls.py]: Done.

**Review by taimoor-ahmed-1 — COMMENTED** [review]
Kindly update PR title and description.

**Comment by madnihamza1841** [comment]
> Kindly update PR title and description.

File: openedx/features/edly/api/v1/urls.py [hunk 2]
```diff
@@ -1,8 +1,14 @@
...
+from openedx.features.edly.api.v1.views.course_enrollments import EdlyCourseEnrollmentViewSet
 from openedx.features.edly.api.v1.views.user_sites import UserSitesViewSet
 
 router = routers.SimpleRouter()
 router.register(r'user_sites', UserSitesViewSet, base_name='user_sites')
+router.register(
+    r'course_enrollment',
+    EdlyCourseEnrollmentViewSet,
+    base_name='course_enrollment'
+    )
```
Comment (taimoor-ahmed-1) [inline openedx/features/edly/api/v1/urls.py:12]:
```
    base_name='course_enrollment`,
)

File: openedx/features/edly/api/v1/views/course_enrollments.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+"""
+Views for course enrollments API
+"""
+from figures.views import CourseEnrollmentViewSet
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from edx_rest_framework_extensions.auth.session.authentication import SessionAuthenticationAllowInactiveUser
+from openedx.core.lib.api.authentication import BearerAuthenticationAllowInactiveUser
+
```
Comment (taimoor-ahmed-1) [inline openedx/features/edly/api/v1/views/course_enrollments.py:8]: remove empty line

File: openedx/features/edly/api/v1/views/course_enrollments.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    **Example Request**
+
+        GET /api/v1/courses/course_enrollment/
+
+    **Response Values**
+
+        If the request is successful, the request returns an HTTP 200 "OK" response.
+    """
+    authentication_classes = (JwtAuthentication, BearerAuthenticationAllowInactiveUser,
+                              SessionAuthenticationAllowInactiveUser,)
```
Comment (taimoor-ahmed-1) [inline openedx/features/edly/api/v1/views/course_enrollments.py:27]:
```
    authentication_classes = (
        JwtAuthentication,
        BearerAuthenticationAllowInactiveUser,
        SessionAuthenticationAllowInactiveUser,
    )

File: openedx/features/edly/api/v1/urls.py (hunk 2 above)
Comment (madnihamza1841) [reply openedx/features/edly/api/v1/urls.py:12]: image

Keeping it consistent with above. Also I don't think it is necessary as it wont reduce any git diff either.

File: openedx/features/edly/api/v1/views/tests/test_course_enrollments.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        """
+        Setup initial test data
+        """
+        super(TestEdlyCourseEnrollmentViewSett, self).setUp()
+        self.request_site = SiteFactory()
+        self.edly_sub_org = EdlySubOrganizationFactory(
+            lms_site=self.request_site,
+            studio_site=self.request_site,
+            preview_site=self.request_site
+            )
```
Comment (taimoor-ahmed-1) [inline openedx/features/edly/api/v1/views/tests/test_course_enrollments.py:29]:
```
    preview_site=self.request_site,
)

Comment (madnihamza1841) [reply openedx/features/edly/api/v1/views/tests/test_course_enrollments.py:29]: Done.

**Review by taimoor-ahmed-1 — APPROVED** [review]
LGTM 👍 
Kindly test changes with correct permissions on dev after merging.

**Pull request merged**